src/services/UserService.ts

```diff
@@ -0,0 +1,37 @@
+import { Service } from '@tsed/common';
+import { Unauthorized } from '@tsed/exceptions';
+import { User } from '../entities/User';
+import { UserRepository } from '../repositories/UserRepository';
+import { UserValidationService } from './UserValidationService';
+
+@Service()
+export class UserService {
+
+	constructor(
+		private userRepository: UserRepository,
+		private userValidationService: UserValidationService) {}
+
+  public async create(user: User): Promise<void> {
+		const validatedUser: User = await this.userValidationService.validateUser(user); 
+		if (await this.isAvailableUsername(validatedUser.username)
+		&& await this.isAvailableEmail(validatedUser.email)) {
+			this.userRepository.save(validatedUser);
+		} 
+	}
+	
+	private async isAvailableUsername(username: string): Promise<boolean> {
+		if (await this.userRepository.findByUsername(username) == undefined) {
+			return true;
+		} else {
+			throw new Unauthorized("Username already taken!");
```

Comment:
After returning a value, the code will not go further. Because of that, the `else` block can be removed and you can keep the exception throw only, Same is true for `availableEmail`. In general:
```
if (someCondition) return true;
else throw new Error();
```
is the same as
```
if (someCondition) return true;
throw new Error();
```